# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.h
# Commit: 69e7c8a0cca3
# Full Hash: 69e7c8a0cca3930210bb22918b5d0f808372ee48
# Author: stransky <stransky@redhat.com>
# Date: 2024-07-07 20:51:50
# Regressor Bug: 1898476
# File Overlap Count: 1
# Description:
#   Bug 1898476 [Wayland] Provide surface lock by GtkCompositorWidget r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D214884
# ==============================================================================

diff -r 480f3f701046 -r 69e7c8a0cca3 widget/gtk/nsWindow.h
--- a/widget/gtk/nsWindow.h	Sun Jul 07 08:46:39 2024 +0000
+++ b/widget/gtk/nsWindow.h	Sun Jul 07 08:46:40 2024 +0000
@@ -42,6 +42,7 @@
 #  include "base/thread.h"
 #  include "WaylandVsyncSource.h"
 #  include "nsClipboardWayland.h"
+#  include "MozContainerSurfaceLock.h"
 #endif
 
 #ifdef MOZ_LOGGING
@@ -421,6 +422,8 @@
 
   static nsWindow* GetFocusedWindow();
 
+  mozilla::UniquePtr<MozContainerSurfaceLock> LockSurface();
+
 #ifdef MOZ_WAYLAND
   // Use xdg-activation protocol to transfer focus from gFocusWindow to aWindow.
   static void TransferFocusToWaylandWindow(nsWindow* aWindow);
